drvregistry/drvregistry.go

```diff
@@ -164,10 +164,10 @@ func (r *DrvRegistry) RegisterDriver(ntype string, driver driverapi.Driver, capa
 	}
 
 	r.Lock()
-	_, ok := r.drivers[ntype]
+	dd, ok := r.drivers[ntype]
 	r.Unlock()
 
-	if ok {
+	if ok && dd.driver.IsBuiltIn() {
```

Comment:
We should atleast add a comment about pluginv2 behaviour, where the older instance of the plugin doesnt get removed from drvRegistry and re-installing a new instance of the plugin overwrites the driverData here.